### ISSUE TITLE
[Bug] Shell MenuItem Text Binding values not updated on Android

### ISSUE THREAD

**Issue by domagojmedo** [xamarin/Xamarin.Forms#9205]
### Steps to Reproduce

1. Bind the Text Property of a MenuItem to a Property in a ViewModel that implements INotifyPropertyChanged.
2. Change the value of that property at runtime.

### Expected Behavior
Text of MenuItem should change

### Actual Behavior
Old value is still shown

### Basic Information

- Version with issue: 4.4.0.991477
- Platform Target Frameworks: 
  - iOS:  
  - Android: 9.0

**Comment by PureWeen** [comment]
@domagojmedo if you set the mode to TwoWay does it update?

**Comment by domagojmedo** [comment]
@PureWeen nope. Setting Title to OneWay works for FlyoutItem, but nothing seems to work for MenuItem Text

**Comment by StephaneDelcroix** [comment]
please provide a reproduction project.

**Comment by domagojmedo** [comment]
Here is the simplest example I could make where MenuItems don't update:

- https://github.com/domagojmedo/ShellBug1

**Comment by PureWeen** [comment]
@domagojmedo I tested your sample and it works great

```XAML
<MenuItem Text="{Binding MenuItemText, Mode=TwoWay}"/>
```

updates 

and all the others that are just oneway don't

**Comment by domagojmedo** [comment]
@PureWeen I tested it again on OnePlus 3T and it's not working, only the first item in flyout menu changes text but that's not MenuItem, that's FlyoutItem

![image](https://user-images.githubusercontent.com/12157224/75015574-34203480-5489-11ea-9cac-8df118c83c76.png)

Green one changes but red one stays the same

**Comment by PureWeen** [comment]
Sorry about that @domagojmedo 

That's really confusing that it doesn't work but probably super obvious why it doesn't once I figure it out :-) 

In the mean time you can replace the template for menuitem with the template defined here
https://docs.microsoft.com/en-us/xamarin/xamarin-forms/app-fundamentals/shell/flyout#default-template-for-flyoutitems-and-menuitems

And define your own binding like so. I tested this against your sample and it works

```XAML
<Shell.MenuItemTemplate>
        <DataTemplate>
            <Grid HeightRequest="{x:OnPlatform Android=50}">
                <VisualStateManager.VisualStateGroups>
                    <VisualStateGroupList>
                        <VisualStateGroup x:Name="CommonStates">
                            <VisualState x:Name="Normal">
                            </VisualState>
                            <VisualState x:Name="Selected">
                                <VisualState.Setters>
                                    <Setter Property="BackgroundColor" Value="#F2F2F2" />
                                </VisualState.Setters>
                            </VisualState>
                        </VisualStateGroup>
                    </VisualStateGroupList>
                </VisualStateManager.VisualStateGroups>
                <Grid.ColumnDefinitions>
                    <ColumnDefinition Width="{x:OnPlatform Android=54, iOS=50}"></ColumnDefinition>
                    <ColumnDefinition Width="*"></ColumnDefinition>
                </Grid.ColumnDefinitions>
                <Image Source="{Binding FlyoutIcon}"
                    VerticalOptions="Center"
                    HorizontalOptions="Center"
                    HeightRequest="{x:OnPlatform Android=24, iOS=22}"
                    WidthRequest="{x:OnPlatform Android=24, iOS=22}">
                </Image>
                <Label VerticalOptions="Center"
                Text="{Binding BindingContext.MenuItemText}"
                FontSize="{x:OnPlatform Android=14, iOS=Small}"
                FontAttributes="Bold" Grid.Column="1">
                    <Label.TextColor>
                        <OnPlatform x:TypeArguments="Color">
                            <OnPlatform.Platforms>
                                <On Platform="Android" Value="#D2000000" />
                            </OnPlatform.Platforms>
                        </OnPlatform>
                    </Label.TextColor>
                    <Label.Margin>
                        <OnPlatform x:TypeArguments="Thickness">
                            <OnPlatform.Platforms>
                                <On Platform="Android" Value="20, 0, 0, 0" />
                            </OnPlatform.Platforms>
                        </OnPlatform>
                    </Label.Margin>
                    <Label.FontFamily>
                        <OnPlatform x:TypeArguments="x:String">
                            <OnPlatform.Platforms>
                                <On Platform="Android" Value="sans-serif-medium" />
                            </OnPlatform.Platforms>
                        </OnPlatform>
                    </Label.FontFamily>
                </Label>
            </Grid>
        </DataTemplate>
    </Shell.MenuItemTemplate>
```

**Comment by domagojmedo** [comment]
No problem, thanks for the workaround

**Comment by ymebrugts** [comment]
I also have this bug right now.

**Comment by ymebrugts** [comment]
Any update on this?

**Comment by thisismyself** [comment]
I experience this issue as well. After I added `x:DataType="MenuItem"` to the DataTemplate, the Binding to property Text of the MenuItem does not work anymore. When removing x:DataType="MenuItem" everything works again, but I also have to remove the x:DataType on the AppShell definition. 

As you can see in the comments, there are some other things not working, not related in to x:DataType.

Xamarin.Forms 4.8.1687.

Some lines of AppShell.xaml:

    <Shell.MenuItemTemplate>
        
        <DataTemplate x:DataType="MenuItem">
            
            <Grid ColumnDefinitions="Auto,Auto">
                
                <Image Grid.Column="0"
                       Margin="15,0,0,0"
                       HeightRequest="24">
                    <Image.Source>
                        
                        <FontImageSource FontFamily="{x:Static constants:MaterialDesignFontConstants.FontFamily}"
                                         Glyph="{x:Static constants:MaterialDesignFontConstants.EmailOutline}"
                                         Color="{x:AppThemeBinding Light={x:Static constants:ColorConstants.InterfaceDarkGray03},
                                                                   Dark={x:Static constants:ColorConstants.InterfaceGray01}}" />
                    </Image.Source>
                </Image>

                <Label Grid.Column="1"
                       Margin="10,10"
                       Text="{Binding Text}"
                       TextColor="{AppThemeBinding Light={x:Static constants:ColorConstants.MaterialBlack},
                                                   Dark={x:Static constants:ColorConstants.InterfaceGray01}}"
                       VerticalTextAlignment="Center">
                    <Label.Triggers>
                        <DataTrigger Binding="{Binding IsEnabled}"
                                     TargetType="Label"
                                     Value="False">
                            <Setter Property="TextColor" Value="{AppThemeBinding Light={x:Static constants:ColorConstants.InterfaceGray03}, Dark={x:Static constants:ColorConstants.InterfaceDarkGray01}}" />
                        </DataTrigger>
                    </Label.Triggers>
                </Label>
            </Grid>
        </DataTemplate>
    </Shell.MenuItemTemplate>

... many FlyoutItems here, no problems with them ....

    <MenuItem Command="{Binding SupportAndFeedbackCommand}"
              Text="{i18n:Translate SupportFeedback}" />

**Comment by ymebrugts** [comment]
I am on Xamarin 5 right now and you should too, Xamarin 5 is way better than all the other releases. The text binding issue seems to be fixed on v5.
